include/brickengine/collision_return_values.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef FILE_COLLISION_RETURN_VALUES_HPP
+#define FILE_COLLISION_RETURN_VALUES_HPP
+
+#include <optional>
+struct CollisionReturnValues{
+    CollisionReturnValues(double spaceLeft, int objectId) : spaceLeft(spaceLeft), objectId(objectId) {}
+    double spaceLeft;
+    int objectId;
+};
+
+struct TriggerReturnValues{
+    TriggerReturnValues(bool isInTrigger, int objectId) : isInTrigger(isInTrigger), objectId(objectId) {}
+    bool isInTrigger;
+    std::optional<int> objectId;
```

Comment:
You don't actually use the optional rn, check [the reference](https://en.cppreference.com/w/cpp/utility/optional) on how to properly use it